### ISSUE TITLE
Release final 1.0 version

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#2758]
For now, I have copied over some of the 'high-level' to-dos from the 0.4 checklist that were not done. We should brainstorm on which of these we want to keep, and what other high-level features we want for 1.0.

**Integration of units:**

* [x] Improve integration of Table columns and quantities (#664, #1762; possible implementation in #1839) - @astrofrog @taldcroft @embray @mhvk
* [x] Improve integration of Table columns with Time and Coordinate objects (#664; sample implementations #1833 and #1835) - @taldcroft @mhvk 
* [ ] <s>Implement masked quantities (#1852) - @adrn @mhvk</s>

**Coordinates, Time, and Observatories:**

* [x] fully implement IAU 2000 trasnformation stack for ICRS -> AltAz @eteq @mhvk
* [ ] <s>automate `IERS` updates (#1697) and other data with expiration dates. @mhvk @taldcroft @eteq</s>
* [ ] <s>barycentric coordinates and velocities, ideally at VLBI precision (if things go quickly enough)</s>

**Stabilize modeling API:**

* [ ] <s>implement support for Quantities(#1464) @nden @embray @wkerzendorf
* [x] implement composite models (including fitting) (#1054) @nden @embray @wkerzendorf
* [x] implement custom_model_2d (#1763) @embray  (really the goal here is just `custom_model` ND)
* [ ] <s>add 'default_fitter', 'last_fitter' and 'fit' method to models (#1330, #979) @nden @embray @wkerzendorf</s>
* [ ] <s>add support for getting uncertainties on parameters (#1827)</s>

**Miscellaneous:**

* [ ] <s>Implement any backward-incompatible API changes needed for the future generalized WCS (even if the full thing isn't implemented) - @nden @perrygreenfield @jehturner</s>

In addition, the following documentation/help-related issues should be addressed:

* [ ] <s>Implement a help system tied to exceptions/warnings - @demetri</s>

Lower priority issues

* [ ] <s>Better integration of FITS tables and VO tables with the Table class - @mdboom @taldcroft @embray</s>

**Comment by astrofrog** [comment]
@eteq @embray - weren't we supposed to include a list of packages that will actually be supported as LTS? (or maybe easier would be to mention packages that won't be supported?). There is a LTS section in the whatsnew, so that would be a good place to mention it.

**Comment by astrofrog** [comment]
Do any of the entries in the stability page need to be updated?

**Comment by eteq** [comment]
@astrofrog - I was thinking the stability page serves for that purpose.  I don't think we wanted to make specific guarantees beyond whatever is there, right?  We can definitely update the LTS description to point to the stability page, though.

**Comment by astrofrog** [comment]
@eteq - so just to clarify, my point was that in the current 'what's new' we say *This means v1.0 will be supported with bug fixes for 2 years from its release* - the question is, in 1.5 years, if the modeling API has changed a lot for example (hopefully it won't, but just as an example), fixing bugs may be more effort than just backporting, because there might be bugs in the old API that don't exist in the new API (so we have to fix bugs in two APIs independently). Of course we agreed to the LTS so to some extent we have committed to the extra level of effort, but on a previous telecon we had discussed that for sub-packages that may still evolve a lot, we would not guarantee to fix all bugs, especially in API-unstable packages. But I think that most sub-packages are now pretty stable in Astropy, so maybe the issue is moot.

**Comment by perrygreenfield** [comment]
Yes, I thought we had agreed that there may be subpackages that may not get LTS support if relatively new.

**Comment by astrofrog** [comment]
So once we merge https://github.com/astropy/astropy/pull/3459, at the moment the following sub-packages would be considered not stable yet:

* ``analytic functions``
* ``io.misc`` (in a way, what's there now is mostly stable, it's just in might grow over time)
* ``modeling``
* ``stats`` (again, what's there at the moment is kind of stable)
* ``utils`` (same, what's there is stable, but might grow)
* ``visualization``
* ``vo`` (could be upgraded to stable since it's been in since 0.4)

So really, we're only talking about maybe not supporting full bug fixes to ``analytic_functions``, ``modeling``, and ``visualization``, but the rest is probably pretty stable.

**Comment by wkerzendorf** [comment]
I would put NDData on that list - the new version has not been in any
release as of yet.

**Comment by pllim** [comment]
`vo.client` and `vo.validator` are stable in the sense that I don't have time to work on it. Is `vo.samp` also stable?

**Comment by pllim** [comment]
No, wait, wait... I notice that `vo` still use `ConfigAlias`. I don't think we should label it as stable until it switches to new config system for realz. What do you think, @mdboom ?

**Comment by mdboom** [comment]
@pllim: Any uses of `ConfigAlias` will already raise deprecation warnings, so aren't considered part of the current API.  And the new config API is already in place.  So I don't think this should be marked as having an unstable API.

**Comment by pllim** [comment]
Okay, thanks!

**Comment by eteq** [comment]
See #3469 for an addition to what's new that points readers to the stability document to know what they can expect LTS for.

**Comment by astrofrog** [comment]
@eteq - I think we are ready for RC2, there are no more PRs for 1.0

**Comment by eteq** [comment]
I missed the window ;) now #3474 needs to finish testing, and we should decide if #3389 goes in 1.0.  After that I'll cut an RC.

**Comment by eteq** [comment]
Note that I added a wiki page at https://github.com/astropy/astropy/wiki/v1.0-RC-testing to allow people to report testing results and announced with the RC2 release.  (we'll see if anyone actually uses it...)

**Comment by astrofrog** [comment]
I think we are good to go apart from https://github.com/astropy/astropy/pull/3512 and #3458. All remaining issues from the RCs are non-critical as far as I can see.

**Comment by astrofrog** [comment]
@embray - I think we are all set to go, and #3515 (updating astropy-helpers) is passing. Can you go ahead and release astropy-helpers 1.0? Then I'm happy to try and do the final 1.0 release for astropy since I have some time this week (unless you wanted to do it).

**Comment by embray** [comment]
Perhaps we should take all the unchecked check boxes on this issue and move them to a "Release final 1.1 version"? :)

**Comment by astrofrog** [comment]
@embray - will do it now

**Comment by astrofrog** [comment]
Unaddressed issues moved to https://github.com/astropy/astropy/issues/3519

**Comment by astrofrog** [comment]
@embray - are we good to go for the release? Shall I give it a try?

**Important note:** once the release is done I will ping the conda folk to make sure they create a new package before we announce it, so that we can advertise conda as an installation/update method. We need to prepare the announcements anyway.

**Comment by jehturner** [comment]
FYI, we also intend to release a new Ureka (1.5) with it after a couple of weeks of further testing.

**Comment by embray** [comment]
@astrofrog If you'd like to do it that's fine.  I'm also happy to do it.  I think as long as we're coordinated on the fact that this is going to happen imminently (so that the proper website updates and announcements can go out concurrently...)

**Comment by astrofrog** [comment]
@embray - ok, if you are happy to do it in the next few hours, then please do it and I'll work on making sure the website and announcements are ready. If you don't have time in the next few hours let me know and I'll do it.

But importantly, once the tar file is ready, we should wait until tomorrow to announce it, to allow the conda people to build a conda package, so we should not send out any emails to astropy or astropy-dev. I'll also update MacPorts.

**Comment by embray** [comment]
@astrofrog So are you saying we should or should not upload it to PyPI?  Or are you saying that for typical users having it on PyPI won't affect them anyways?

**Comment by astrofrog** [comment]
We should upload it to PyPI now.

If anyone happens to pip install between now and the announcement, then it's fine. But we don't want to put out the big announcement, have everyone who is using anaconda use pip to update because it will mess things up in future. So it's just the announcement that needs to wait for a day after the tar file is on PyPI, so that we can mention 'conda update astropy' as a possible update mechanism.

**Comment by astrofrog** [comment]
Just for info, changes to the website including announcement are here: https://github.com/astropy/astropy.github.com/pull/51

**Comment by zblz** [comment]
I see that the 1.0 conda binary package is already available for numpy 1.9, but for 1.8, 1.7, and 1.6 the conda package is an older version (0.4.1, 0.4.0, and 0.4.0, respectively). Do you know if astropy will be updated for all numpy versions? It would be really useful for travis testing, otherwise it is impossible to test features that require astropy>=1.0 with older numpy.

**Comment by astrofrog** [comment]
@zblz - I'm currently working on building precisely what you ask :) We have our own conda channel (https://binstar.org/astropy-ci-extras) for this and I've just launched the builds for astropy 0.3, 0.4, and 1.0 (the repo used to build these is https://github.com/astropy/conda-builder). More soon.

**Comment by astrofrog** [comment]
@zblz - basically, once these packages are ready you should do e.g.

```
conda install -c astropy-ci-extras numpy=1.6 astropy=1.0
```

**Comment by zblz** [comment]
@astrofrog - Thanks!

**Comment by astrofrog** [comment]
@zblz - some of the files have started to appear so you can start testing this if you like:

https://binstar.org/astropy-ci-extras/astropy/files

**Comment by zblz** [comment]
The ones that have appeared (1.7 and 1.6) are working fine: https://travis-ci.org/zblz/naima/jobs/51348825

**Comment by astrofrog** [comment]
@zblz - I think all the linux packages are done. Are there any missing for your purposes?

**Comment by zblz** [comment]
@astrofrog - No, everything is working. Thanks again!

**Comment by embray** [comment]
That reminds me, I also need to post wheels for Windows to PyPI.

**Comment by mhvk** [comment]
This can be closed! Yay!
